sound/soc/sof/pcm.c

```diff
@@ -533,6 +535,7 @@ static int sof_pcm_close(struct snd_pcm_substream *substream)
 	sdev->pcm_active--;
 	if (spcm->stream[substream->stream].avail_at_lp)
 		sdev->lp_pcm_active--;
+	schedule_work(&sdev->d0ix_work);
```

Comment:
so what is missing in this series is something that will run after e.g each IPC.
This is not just about opening/closing streams, we need to keep the DSP active while we are doing IPCs.

Reply:
Yes, I planed to do that with IPC with registry part(need alignment with FW), it is in the to-do list also.